server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java

```diff
@@ -215,8 +215,8 @@ public void onFailure(Exception e) {
         }
     }
 
-
     public static class Status implements Task.Status {
```

Comment:
do we need this one? How bad would it be to remove it and let the task implementation add one if they need to?

Reply:
It's the default status implementation for allocated persistent task and it just shows the internal state (started/completed/pending cancel) of the persistent task.

I think that we don't really need it and I'd prefer to see it removed. But this change was already big, and removing this class means to take care of some backward compatibility (this status is used as part of the TaskInfo object which is serialized over the network when using the Tasks API and also potentially persisted in TaskResult).

Reply:
Ok. makes sense